src/cargo/ops/cargo_add/mod.rs

```diff
@@ -124,11 +126,71 @@ pub fn add(workspace: &Workspace<'_>, options: &AddOptions<'_>) -> CargoResult<(
                 inherited_features.iter().map(|s| s.as_str()).collect();
             unknown_features.extend(inherited_features.difference(&available_features).copied());
         }
+
         unknown_features.sort();
+
+        let mut activated: IndexSet<_> =
+            dep.features.iter().flatten().map(|s| s.as_str()).collect();
+        if dep.default_features().unwrap_or(true) {
+            activated.insert("default");
+        }
         if !unknown_features.is_empty() {
-            anyhow::bail!("unrecognized features: {unknown_features:?}");
+            let (activated, deactivated) = dep.features();
+            let deactivated = deactivated
+                .iter()
+                .filter(|f| !unknown_features.contains(f))
+                .collect::<Vec<_>>();
+            let activated = activated
+                .iter()
+                .filter(|f| !unknown_features.contains(f))
+                .collect::<Vec<_>>();
+            let mut message = format!(
+                "unrecognized feature{} for crate {}: {}\n",
+                if unknown_features.len() == 1 { "" } else { "s" },
+                dep.name,
+                unknown_features.iter().format(", "),
+            );
+            if activated.is_empty() && deactivated.is_empty() {
+                write!(message, "no features available for crate {}", dep.name)?;
```

Comment:
We should probably have a test that covers this case as well

Reply:
I added this with `cargo_add::features_unknown::features_unknown_empty` not sure if it should really be added as it's own folder for `cargo_test`

Reply:
Please use one test case per `mod.rs`.